proposed/2022/CPMDotnetCLISupport.md

```diff
@@ -0,0 +1,134 @@
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1"/>
+    </ItemGroup>
+</Project>
+```
+The .csproj file: <br>
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+    <PropertyGroup>
+        <TargetFramework>net6.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Newtonsoft.Json"/>
+    </ItemGroup>
+</Project>
+```
+
+### 2. The package reference does exist
+If the package already exists in `Directory.packages.props` the version should be updated in `Directory.packages.props`. The package version should either be the latest version or the one specified in the CLI command. The `<PackageReference>` in the project file should not change.
+
+#### Before `add package` is executed:<br>
+```xml
+<Project>
+    <PropertyGroup>
+        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="12.0.1"/>
+    </ItemGroup>
+</Project>
+```
+
+#### After `add package` is executed:<br>
+```xml
+<Project>
+    <PropertyGroup>
+        <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    </PropertyGroup>
+    <ItemGroup>
+    <PackageVersion Include="Newtonsoft.Json" Version="13.0.1"/>
+    </ItemGroup>
+</Project>
+```
+
+### 3. There are multiple `Directory.packages.props` files
+In the case that there are multiple `Directory.packages.props` files in the repo, the props file that is closest must be considered.
+
+```xml
+Repository
+ |-- Directory.Packages.props
+ |-- Solution1
+    |-- Directory.Packages.props
+    |-- Project1
+ |-- Solution2
+    |-- Project2
+```
+
+In the above example, the following scenarios are possible:
+1. Project1 will evaluate the Directory.Packages.props file in the Repository\Solution1\ directory.
+2. Project2 will evaluate the Directory.Packages.props file in the Repository\ directory.
+
+***Sourced from https://devblogs.microsoft.com/nuget/introducing-central-package-management/
+
+### Other Scenarios that can be considered
+
+While the above cases must be considered to solve the issue, there are a few other cases that can be considered to make the product more usable:
+
+1. If the project has the <ManagePackageVersionsCentrally> set to true, but the `Directory.packages.props` file has not been created yet the file should be created. At this point, package references can be updated according to the cases discussed above.
+2. If the `Directory.packages.props` file has been created but the <ManagePackageVersionsCentrally> property has not been set to true, it should be set to true.
+
+### Unresolved Questions
+1. If there are `Directory.packages.props` files inherit other props files, which props file should be modified when `add package` is executed?
```

Comment:
Most likely it would be the closest one enabled CPM. Of course, calculating that one could be tricky.

Reply:
After discussing with Jon, we decided it would be best to consider all imported files when adding package references. This way if there is a package reference in an imported file that needs to be updated, then it can be updated accordingly.